### PR TITLE
[SPARK-15136][PySpark][DOC] Fix links to sphinx style and add a default param doc note

### PR DESCRIPTION
**Pull request by holdenk**
## What changes were proposed in this pull request?

PyDoc links in ml are in non-standard format. Switch to standard sphinx link format for better formatted documentation. Also add a note about default value in one place. Copy some extended docs from scala for GBT
## How was this patch tested?

Built docs locally.

### CODE REVIEW

**Comment by holdenk** [comment]
cc @yanboliang

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -454,7 +454,8 @@ class TreeClassifierParams(object):
     impurity = Param(Params._dummy(), "impurity",
                      "Criterion used for information gain calculation (case-insensitive). " +
                      "Supported options: " +
-                     ", ".join(supportedImpurities), typeConverter=TypeConverters.toString)
+                     ", ".join(supportedImpurities) + ". (Default: gni)",
```
Comment (srowen) [inline python/pyspark/ml/classification.py]: gini not gni. Does the default have to be explicitly documented like this? I didn't see other params doing it but I just looked in this file

Comment (holdenk) [inline python/pyspark/ml/classification.py]: It hasn't been on the python side, but the Scaladoc includes mention of the default values so I've been adding it as I go through them anyways (see https://github.com/apache/spark/pull/12914 for the shared params case)

**Comment by yanboliang** [comment]
I like it! @holdenk Could you make change for all links in spark.ml docs? I found there are links in ```regression.py```, ```recommendation.py``` and ```feature.py``` also need update. Thanks!

File: python/pyspark/ml/classification.py [hunk 2]
```diff
@@ -353,7 +353,7 @@ def roc(self):
         Returns the receiver operating characteristic (ROC) curve,
         which is an Dataframe having two fields (FPR, TPR) with
         (0.0, 0.0) prepended and (1.0, 1.0) appended to it.
-        Reference: http://en.wikipedia.org/wiki/Receiver_operating_characteristic
+        `Wikipedia reference <http://en.wikipedia.org/wiki/Receiver_operating_characteristic>`_
```
Comment (yanboliang) [inline python/pyspark/ml/classification.py]: Use ```seealso``` here? For your reference:
```.. seealso:: `AFT Model <https://en.wikipedia.org/wiki/Accelerated_failure_time_model>`_```

**Comment by holdenk** [comment]
Happy to, I was going to leave those files for the people working in them during QA but I can update them as well.

File: python/pyspark/ml/classification.py [hunk 3]
```diff
@@ -454,7 +454,8 @@ class TreeClassifierParams(object):
     impurity = Param(Params._dummy(), "impurity",
                      "Criterion used for information gain calculation (case-insensitive). " +
                      "Supported options: " +
-                     ", ".join(supportedImpurities), typeConverter=TypeConverters.toString)
+                     ", ".join(supportedImpurities) + ". (Default: gini)",
```
Comment (MLnick) [inline python/pyspark/ml/classification.py]: @holdenk could we leave this change out of this PR and keep it only for links? When we make a final call on approach for #12914 we can revisit.

Comment (holdenk) [inline python/pyspark/ml/classification.py]: I can revert this if we want or we could just wait to decide on https://github.com/apache/spark/pull/12914 first

Comment (MLnick) [inline python/pyspark/ml/classification.py]: I think the rest of this PR can move ahead and get merged, so let's just revert this line and follow up later.

**Comment by srowen** [comment]
That LGTM

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**